### PR TITLE
Port TwinkleFox effect from FastLED

### PR DESCRIPTION
**Pull request by Aircoookie**
This is my absolute favorite LED effect, so I think it should not be missing in this library!
Implemented as Custom effect. Standalone, FastLED library is not required.
If colors[0] is black, rainbow colors will be used!
Speed 0-65535, higher value is slower

 TwinkleFOX by Mark Kriegsman: https://gist.github.com/kriegsman/756ea6dcae8e30845b5a

Hope you enjoy :)

### CODE REVIEW

File: src/custom/TwinkleFox.h [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+#ifndef Twinklefox_h
+#define Twinklefox_h
+
+#include <WS2812FX.h>
+
+// If COOL_LIKE_INCANDESCENT is set to 1, colors will
+// fade out slighted 'reddened', similar to how
+// incandescent bulbs change color as they get dim down.
+#define COOL_LIKE_INCANDESCENT 1
+
```
Comment (moose4lord) [inline src/custom/TwinkleFox.h]: Is this used anywhere?

File: src/custom/TwinkleFox.h [hunk 2]
```diff
@@ -0,0 +1,209 @@
+/*
+  TwinkleFOX by Mark Kriegsman: https://gist.github.com/kriegsman/756ea6dcae8e30845b5a
+  If colors[0] is black, rainbow colors will be used
+  Speed 0-65535, higher value is slower
+  
+  Keith Lord - 2018
```
Comment (moose4lord) [inline src/custom/TwinkleFox.h]: Replace with your name.

File: src/custom/TwinkleFox.h [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+  LICENSE
+  The MIT License (MIT)
+  Copyright (c) 2018  Keith Lord
```
Comment (moose4lord) [inline src/custom/TwinkleFox.h]: Replace with your name.

**Comment by moose4lord** [comment]
Oops. First time I've commented on a pull request. I didn't mean for each of my remarks to be a separate comment. Oh well.
I like the effect. The code is pretty messy though. Some of the variables have poor names and the code could use better/more comments to describe what is going on. I also wish it could take advantage of all colors in the segment's colors[] array.

**Comment by Aircoookie** [comment]
Thanks for the feedback :)
I will get back to you with an improved version soon!

**Comment by moose4lord** [comment]
Hello @Aircookie, I showed the TwinkleFox custom effect to my wife and now she wants to include it in our Christmas repertoire. So I've refactored your version of TwinkleFox to remove the FastLED cruft and implement it the WS2812FX "way". If you want to try it out, it's in my WS2812FX fork [here](https://github.com/moose4lord/WS2812FX/blob/master/src/custom/TwinkleFox.h).

**Comment by moose4lord** [comment]
I've added an implementation of TwinkleFox to WS2812FX's custom effects.

**Comment by Aircoookie** [comment]
@moose4lord sorry for not getting back to you on this earlier, have been quite busy.
I like the new syntax, it looks very clean!
Thank you for adding this effect to the libraty :)

**Pull request closed without merging**